cdap-api/src/main/java/co/cask/cdap/api/schedule/ScheduleSpecification.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright © 2015 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.api.schedule;
+
+import co.cask.cdap.api.workflow.ProgramNameTypeInfo;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Specification for {@link Schedule}.
+ */
+public class ScheduleSpecification {
+  private final Schedule schedule;
+  private final List<ProgramNameTypeInfo> programList;
+
+  public ScheduleSpecification(Schedule schedule) {
+    this.schedule = schedule;
+    this.programList = new ArrayList<ProgramNameTypeInfo>();
```

Comment:
Any specific reason why an ArrayList ? Are you going to access the list with index anywhere ? I don't think so. 
Since we are just adding to the end and iterating over them in sequence a LinkedList will be better.